docs/migration-guides/prometheus-migration.md

```diff
@@ -0,0 +1,328 @@
+
+## Deploying the sidecar rsync server
+
+In order to access via rsync to the Prometheus TSDB snapshots, it will be necessary to deploy a rsync server with a minimal configuration to make it so.
+
+This way, you can use the following jsonnet snippet to add that sidecar container:
+
+```jsonnet
+// target-add-rsync.jsonnet: add rsync sidecar container, exposing
+// <volume_name> below via rsync, by overloading saved "target.json"
+
+// NOTE: We're not assuming a particular instrumentation tool (e.g ksonnet, kubecfg via jsonnet)
+// but rather doing an override from original sts / deploy output json
+
+// 1) Save existing deployment/sts to target.json:
+//   kubectl get deployments -n NAMESPACE NAME -o target.json
+local target = import "target.json";
+
+// volume_name must be the name of existing volume name inside target's podSpec
+local volume_name = "data";
+
+// 2) show/diff/update resulting composition ('-J lib' where lib is the path to kube.libsonnet)
+//   kubecfg -J lib show target-add-rsync.jsonnet
+
+// For testing purposes, You can fake above by creating a deploy with an emptyDir volume:
+//    kubectl run --image=nginx:latest --overrides '{"spec":{"template":{"spec":{ "volumes": [{"name": "data", "emptyDir": {}}]}}}}' nginx-tgt
+//    kubectl get deploy nginx-tgt -ojson > target.json
+
+local kube = import "./kube.libsonnet";
+local rsync_conf_b64 = std.base64(
+  |||
+    [data]
+    path = /data
+    list = false
+    use chroot = false
+    read only = true
+  |||
+);
+
+local rsync_container_name = "sidecar-rsync";
+
+// sidecar-rsync container (mounting <volume_name>) to be exposed via rsync
+local rsync_container = kube.Container(rsync_container_name) {
+  image: "alpine:3.8",
+  args: [
+    "sh",
+    "-c",
+    |||
+      apk update && apk add rsync curl
+      echo '%s' | base64 -d > /rsync.conf
+      rsync -vvv --daemon --no-detach --config=/rsync.conf
+    ||| % [rsync_conf_b64],
+  ],
+  volumeMounts_+: {
+    [volume_name]: { mountPath: "/data" },
+  },
+};
+
+local containers_obj(containers) = {
+  [container.name]: container for container in containers
+};
+
+local stack = {
+  
+  target_with_sidecar: target {
+    spec+: {
+      template+: {
+        spec+: {
+          // convert containers array to obj for easier jsonnet
+          // handling: in particular, if we just added the new container
+          // to existing containers[] array, we wouldn't be able to re-run
+          // this, as we'd get duplicated containers.
+          // By referring the added container via rsync_container_name, we
+          // can easily override it.
+
+          local containers = super.containers,
+          containers: kube.PodSpec {
+            default_container:: containers[0].name,
+            // create or override(existing) rsync_container
+            containers_+: containers_obj(containers) {
+              [rsync_container_name]: rsync_container,
+              ["prometheus"]+:{
```

Comment:
Please add a comment regarding required naming, i.e. main existing
container _needs_ to be named `"prometheus"`, although I'd add a
local variable on-top for easier user editing.